src/value-converters/highlight-text-value-converter.ts

```diff
@@ -36,7 +36,7 @@ export class HighlightTextValueConverter {
             regExp = new RegExp(this._htmlSanitizer.sanitize(regExp.source), regExp.flags);
             const match = value.match(regExp);
             if (match && match.length) {
-                value = value.replace(regExp, '<mark>$1</mark>');
+                value = value.replace(regExp, '<mark>$&</mark>');
```

Comment:
I changed this because I noticed that for any RegExp which doesn't match **exactly one bracketed group**, this replacement does not work. 

Fiddling around with the code editor on here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#try_it, I found that `$1` is the replacement pattern for "first group" and `$&` is the pattern for "all groups", which is the right one I believe.

**Examples:**
Using `$1` with a RegExp with no group matchers:
```
const p = 'The quick brown fox jumps over the lazy dog. If the dog reacted, was it really lazy?';

const regex = /dog/g;
console.log(p.replace(regex, '<mark>$1</mark>'));
```
Output:
```
"The quick brown fox jumps over the lazy <mark>$1</mark>. If the <mark>$1</mark> reacted, was it really lazy?"
```

Using `$&` with a RegExp with no group matchers:
```
const p = 'The quick brown fox jumps over the lazy dog. If the dog reacted, was it really lazy?';

const regex = /dog/g;
console.log(p.replace(regex, '<mark>$&</mark>'));
```
Output:
```
"The quick brown fox jumps over the lazy <mark>dog</mark>. If the <mark>dog</mark> reacted, was it really lazy?"
```

Using `$1` with a RegExp with two group matchers:
```
const p = 'The quick brown fox jumps over the lazy dog. If the dog reacted, was it really lazy?';

const regex = /(dog)|(fox)/g;
console.log(p.replace(regex, '<mark>$1</mark>'));
```
Output:
```
"The quick brown <mark></mark> jumps over the lazy <mark>dog</mark>. If the <mark>dog</mark> reacted, was it really lazy?"
```

Using `$&` with a RegExp with two group matchers:
```
const p = 'The quick brown fox jumps over the lazy dog. If the dog reacted, was it really lazy?';

const regex = /(dog)|(fox)/g;
console.log(p.replace(regex, '<mark>$&</mark>'));
```
Output:
```
"The quick brown <mark>fox</mark> jumps over the lazy <mark>dog</mark>. If the <mark>dog</mark> reacted, was it really lazy?"
```